### PR TITLE
Replace OpenTracing error logs with SignalFx tags

### PR DESCRIPTION
**Pull request by owais**
Replace OpenTracing error logs with SignalFx tags

This PR changes how errors are recorded on spans. It uses the new
SignalFx semantic convention and records the errors as attributes
instead of logs.

In addition to updating errors, also fixes multiple broken test and lint
jobs in CircleCI.

### CODE REVIEW

File: contrib/google.golang.org/grpc.v12/go.mod [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module github.com/signalfx/signalfx-go-tracing/contrib/google.golang.org/grpc.v12
```
Comment (rmfitzpatrick) [inline contrib/google.golang.org/grpc.v12/go.mod:1]: This is only to be copied from `.circleci` during test runs, correct?

Comment (owais) [reply contrib/google.golang.org/grpc.v12/go.mod:1]: Oh yes. Need to remove this.

**Comment by owais** [comment]
@rmfitzpatrick PTAL

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,69 @@
+module github.com/signalfx/signalfx-go-tracing
```
Comment (rmfitzpatrick) [inline go.mod:1]: Is this intentional?  Upstream isn't using mod and I don't believe we want to declare any target dependencies.

Comment (rmfitzpatrick) [reply go.mod:1]: May be worthwhile to add these to .gitignore

Comment (owais) [reply go.mod:1]: Not intentional. Had to run tests locally and forgot to remove.

Comment (owais) [reply go.mod]: Fixed.

**Pull request merged**